test/algorithms/check_validity.hpp

```diff
@@ -13,22 +13,38 @@
 
 #include <boost/geometry/algorithms/is_valid.hpp>
 
+template<typename Geometry>
+inline bool input_is_valid(std::string const& case_id, std::string const& subcase,
+                         Geometry const& geometry)
+{
+    std::string message;
+    bool const result = bg::is_valid(geometry, message);
+    if (! result)
+    {
+        std::cout << "WARNING: " << case_id << " Input [" << subcase
+                  << "] is not considered as valid ("
+                  << message << ") this can cause that output is invalid: "
+                  << case_id << std::endl;
```

Comment:
You should probably use `BOOST_TEST_MESSAGE` here instead.

Reply:
I don't know it yet, thanks.
I'll merge this (it's consistent to previous behavior) but will look at this in a later phase.